plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveUpdatablePageSource.java

```diff
@@ -171,8 +179,9 @@ Block createRowIdBlock(int positionCount)
     {
         long[] rowIds = new long[positionCount];
         for (int index = 0; index < positionCount; index++) {
-            rowIds[index] = insertRowCounter++;
+            rowIds[index] = initialRowId + insertRowCounter++;
         }
+        checkState(maxNumberOfRowsPerSplit > insertRowCounter, "Trying to insert too many rows in a single split");
```

Comment:
Add `maxNumberOfRowsPerSplit` to error message. Also make it throw `TrinoException` with `GENERIC_INSUFFICIENT_RESOURCES` error code

Reply:
ok